File: First Unique Character in a String.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        dp = {}
        n = len(s)
        for i, c in enumerate(s):
            if c in dp:
                dp[c] = -1
            else:
                dp[c] = i
        minn = n
        for c, i in dp.items():
            if i != -1:
                minn = min(minn, i)
        return minn if minn != n else -1

File: First Unique Character in a String.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char, int> umap;
        for(char& c : s) {
            if (umap.find(c) == umap.end()) {
                umap.insert(make_pair(c, 1));
            } else {
                umap.at(c)++;
            }
        }

        int i = 0;
        for(char& c : s) {
            if (umap.at(c) == 1) return i;
            i++;
        }
        return -1;
    }
};
